File: Venda-distribuida-java/src/main/java/br/ufc/venda/client/esqueleto/EsqueletoBroadcastCliente.java
package br.ufc.venda.client.esqueleto;

import com.google.gson.Gson;

import br.ufc.venda.client.dao.ClienteDAO;
import br.ufc.venda.model.Cliente;
import br.ufc.venda.model.Resposta;

public class EsqueletoBroadcastCliente {
	
	private ClienteDAO cdao;
	private Gson gson;
	
	public EsqueletoBroadcastCliente(){
		cdao = new ClienteDAO();
		gson = new Gson();
	}
	
	public String refresh(String args){
		Cliente cliente = gson.fromJson(args, Cliente.class); 
		Resposta resposta = new Resposta();
		if(cdao.salvar(cliente)){
			resposta.setCodigo(Resposta.SUCCESS);
			resposta.setResposta("true");
		}else{
			resposta.setCodigo(Resposta.ERROR);
			resposta.setResposta("false");
		}
		return gson.toJson(resposta);
	}
	
	public String rollback(String args){
		Cliente cliente = gson.fromJson(args, Cliente.class); 
		Resposta resposta = new Resposta();
		if(cdao.excluir(cliente)){
			resposta.setCodigo(Resposta.SUCCESS);
			resposta.setResposta("true");
		}else{
			resposta.setCodigo(Resposta.ERROR);
			resposta.setResposta("false");
		}
		return gson.toJson(resposta);
	}
	
}

File: Venda-distribuida-python/esqueleto/EsqueletoBroadcastCliente.py
#-*-coding:utf-8 -*-

import json

from dao.ClienteDAO import ClienteDAO

class EsqueletoBroadcastCliente():

    __SUCCESS = 200
    __ERROR = 500

    def __init__(self):
        self.cdao = ClienteDAO()

    def refresh(self, args):
        cliente = json.loads(args)
        resposta = None
        if self.cdao.salvar(cliente):
            resposta = {"codigo":self.__SUCCESS, "reposta":"true"}
        else:
            resposta = {"codigo": self.__ERROR, "reposta": "false"}
        return json.dump(resposta)

    def rollback(self,args):
        cliente = json.loads(args)
        resposta = None
        if self.cdao.excluir(cliente):
            resposta = {"codigo": self.__SUCCESS, "reposta": "true"}
        else:
            resposta = {"codigo": self.__ERROR, "reposta": "false"}
        return json.dump(resposta)